lib/rust/parser/tests/parse.rs

```diff
@@ -0,0 +1,236 @@
+    test(&bool, bool_ast);
+}
+
+#[test]
+#[ignore]
+fn type_definition_geo() {
+    let geo = [
+        "type Geo",
+        "    Circle",
+        "    radius : Float",
+        "    Rectangle",
+        "    width  : Float",
+        "    height : Float",
+    ].join("\n");
+    let geo_ast = block![()];
+    test(&geo, geo_ast);
+}
+
+#[test]
+#[ignore]
+fn type_definition_option() {
+    let option = [
+        "type Option a",
+        "    Some elem:a",
+        "    None",
+    ].join("\n");
+    let option_ast = block![()];
+    test(&option, option_ast);
+}
+
+#[test]
+fn assignment_simple() {
+    test("foo = 23", block![(Assignment (Ident foo) "=" (Number 23))]);
+}
+
+#[test]
+fn function_inline_simple_args() {
+    test("foo a = 23", block![(Function foo #((Ident a)) "=" (Number 23))]);
+    test("foo a b = 23", block![(Function foo #((Ident a) (Ident b)) "=" (Number 23))]);
+    test(
+        "foo a b c = 23",
+        block![(Function foo #((Ident a) (Ident b) (Ident c)) "=" (Number 23))],
+    );
+}
+
+#[test]
+fn function_block_noargs() {
+    test("foo =", block![(Function foo #() "=" ())]);
+}
+
+#[test]
+fn function_block_simple_args() {
+    test("foo a =", block![(Function foo #((Ident a)) "=" ())]);
+    test("foo a b =", block![(Function foo #((Ident a) (Ident b)) "=" ())]);
+    test("foo a b c =", block![(Function foo #((Ident a) (Ident b) (Ident c)) "=" ())]);
+}
+
+
+
+// ====================
+// === Test Support ===
+// ====================
+
+use enso_metamodel::meta;
+use enso_reflect::Reflect;
+use std::collections::HashSet;
+
+/// Given a block of input Enso code, test that:
+/// - The given code parses to the AST represented by the given S-expression.
+/// - The AST pretty-prints back to the original code.
+///
+/// The S-expression format is as documented for [`enso_metamodel::meta::sexp`], with some
+/// postprocessing:
+/// - For concision, field names are stripped (as if all structs were tuple structs).
+/// - Most token types are represented as their contents, rather than as a token struct. For
+///   example, a `token::Number` may be represented like: `sexp![10]`, and a `token::Ident` may look
+///   like `sexp![foo]`.
+fn test(code: &str, expect: lexpr::Value) {
+    let ast = enso_parser::Parser::new().run(code);
+    let ast_sexp = to_sexp(&ast, code);
+    assert_eq!(ast_sexp.to_string(), expect.to_string());
+    assert_eq!(ast.code(), code);
+}
+
+
+
+// =====================
+// === S-expressions ===
+// =====================
+
+/// Produce an S-expression representation of the input AST type.
+pub fn to_sexp<T>(value: &T, code: &str) -> lexpr::Value
+where T: serde::Serialize + Reflect {
+    use bincode::Options;
+    let bincoder = bincode::DefaultOptions::new().with_fixint_encoding();
+    let value_ = bincoder.serialize(&value).unwrap();
+
+    let (graph, rust_to_meta) = enso_metamodel::rust::to_meta(value.reflect_type());
+    let ast_ty = rust_to_meta[&value.reflect_type().id];
+
```

Comment:
this fn is huge. are we able to split it down to several funcs or is it hard? If the latter - could you at least divide it into a few "sections" with comments insidde?